addons/stock_delivery/models/stock_picking.py

```diff
@@ -29,6 +37,22 @@ def _compute_weight_uom_name(self):
     return_label_ids = fields.One2many('ir.attachment', compute='_compute_return_label')
     destination_country_code = fields.Char(related='partner_id.country_id.code', string="Destination Country")
 
+    @api.depends('partner_id', 'carrier_id', 'delivery_method')
```

Comment:
if carrier_id changed than delivery_method is recomputed, so i believe we do not need it here, wdyt?
```suggestion
    @api.depends('partner_id', 'delivery_method')
```